core/src/main/java/com/yahoo/oak/EntrySet.java

```diff
@@ -250,26 +245,13 @@ private long getEntryArrayFieldLong(int intStartFieldIdx, OFFSET offset) {
         return UNSAFE.getLong(entries, arrayOffset);
     }
 
-    /**
-     * getEntryArrayFieldLongVolatile atomically reads two integers field of the entries array.
-     * Should be used with OFFSET.VALUE_REFERENCE and OFFSET.KEY_REFERENCE
-     * The concurrency is ensured due to memory fence as part of "volatile"
-     */
-    private long getEntryArrayFieldLongVolatile(int intStartFieldIdx, OFFSET offset) {
-        assert offset == OFFSET.VALUE_REFERENCE || offset == OFFSET.KEY_REFERENCE;
-        long arrayOffset =
-                Unsafe.ARRAY_INT_BASE_OFFSET + (intStartFieldIdx + offset.value) * Unsafe.ARRAY_INT_INDEX_SCALE;
-        assert arrayOffset % 8 == 0;
-        return UNSAFE.getLongVolatile(entries, arrayOffset);
-    }
-
     /**
      * setEntryFieldInt sets the integer field of specified offset to 'value'
      * for given integer index in the entry array
      */
     private void setEntryFieldInt(int intFieldIdx, OFFSET offset, int value) {
         assert intFieldIdx + offset.value >= 0;
-        entries[intFieldIdx + offset.value] = value; // used for NEXT and VALUE_VERSION
+        entries[intFieldIdx + offset.value] = value; // used for NEXT
     }
 
     private void setEntryFieldLong(int item, OFFSET offset, long value) {
```

Comment:
Similar as above, it is indeed used only within writeKey, because value reference is always set with CAS. But I would like to keep this option, so we can use it for alternative other assignments. For example we may need it for single-thread oak variation if-to-be-implemented...